src/test/java/com/fasterxml/jackson/databind/ser/GenericTypeSerializationTest.java

```diff
@@ -164,6 +167,74 @@ public static Attributes2821 dummyMethod(Map attributes) {
         }
     }
 
+    public static class WrapperWithFactoryMethod<T> {
+        private final List<T> values;
+
+        private WrapperWithFactoryMethod(List<T> values) {
+            this.values = requireNonNull(values, "values is null");
+        }
+
+        @JsonProperty
+        public List<T> getValues() {
+            return values;
+        }
+
+        @JsonCreator
+        public static <T> WrapperWithFactoryMethod<T> fromValues(@JsonProperty("values") List<T> values) {
```

Comment:
Please note that `T` here is NOT related to type variable `T` for class, and as such can not realistically be used as a creator method. It just happens to have same name, but being static method is not part of the class or bindings in that sense.